### PR TITLE
Support dropout

### PR DESCRIPTION
**Pull request by wu-haoze**
Also tested on [NN_rul_full_window_20.onnx](https://github.com/loonwerks/vnncomp2022/blob/main/onnx/NN_rul_full_window_20.onnx) and it seems to work.

### CODE REVIEW

**Review by MatthewDaggitt — CHANGES_REQUESTED** [review]
I don't think that always throwing this off to the identity node parsing is a good approach as we might (silently!) incorrectly parse the network when the `training_mode` input is set to true. 
Looking at the [documentation](https://github.com/onnx/onnx/blob/main/docs/Operators.md#Dropout), we should test for the presence of the third input, and throw a meaningful user error if it is set to true. Only if it is not present, or set to false should we delegate to parsing it as an identity.

**Comment by wu-haoze** [comment]
> I don't think that always throwing this off to the identity node parsing is a good approach as we might (silently!) incorrectly parse the network when the `training_mode` input is set to true. Looking at the [documentation](https://github.com/onnx/onnx/blob/main/docs/Operators.md#Dropout), we should test for the presence of the third input, and throw a meaningful user error if it is set to true. Only if it is not present, or set to false should we delegate to parsing it as an identity node.

Sure. For that, I extended the OnnxParser a little further to handle Int32, since Bool is internally stored as that according to [this](https://github.com/onnx/onnx/blob/606ee9df9580872e34701791b8fbe433cefa6916/onnx/onnx.proto3#L576-L583)  (is this the right file to look at?)

**Review by MatthewDaggitt — CHANGES_REQUESTED** [review]

File: resources/onnx/test_dropout.onnx
[the remark this answers is not in the document]
Comment (wu-haoze) [reply resources/onnx/test_dropout.onnx:1]: The file is used in maraboupy/test/test_onnx.py to pass the python code coverage test. I tried to use the dropout unit models in the model zoo but on the CI, onnxruntime throws an error on those layer zoo models for python version 3.8 for some reason. I think this should be fixed in a separate PR. Perhaps some further amendment to the onnx_layer_zoo.py is needed.

File: resources/onnx/layer-zoo/onnx-layer-zoo.py [hunk 1]
```diff
@@ -24,7 +25,11 @@ def make_network(name, node, input_shape, output_shape, aux_nodes):
     model = onnx.helper.make_model(graph, producer_name=producer_name)
     print(f"Generated {name}.onnx")
     output_dir = os.path.dirname(sys.argv[0])
-    onnx.save(model, os.path.join(output_dir ,f"{name}.onnx"))
+    network_dir = os.path.join(output_dir ,f"{name}.onnx")
```
[the remark this answers is not in the document]
Comment (wu-haoze) [reply resources/onnx/layer-zoo/onnx-layer-zoo.py:28]: Changed to network_path

File: resources/onnx/test_dropout.onnx
[the remark this answers is not in the document]
Comment (wu-haoze) [reply resources/onnx/test_dropout.onnx:1]: See this run https://github.com/NeuralNetworkVerification/Marabou/actions/runs/8147526449/job/22270758583
I spent some effort fixing the issue but to no avail. :(

**Review by MatthewDaggitt — APPROVED** [review]

File: resources/onnx/test_dropout.onnx
[the remark this answers is not in the document]
Comment (MatthewDaggitt) [reply resources/onnx/test_dropout.onnx:1]: Okay fair enough. We should probably open an issue for that...

**Pull request merged**